mautic/mautic
# Fix email stat

| Q  | A
| --- | ---
| Bug fix? | x
| New feature? | -
| Related user documentation PR URL | -
| Related developer documentation PR URL | -
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/3461
| BC breaks? | -
| Deprecations? | -

- [x] Bug displaying uncorrect number of clicked in email overview graph
- [x] Add email clicked email/view
- [x] Add email clicked in report
- [x] Add email clicked ratio in report
- [x] Insure migration scheme on update to recalculate clicked email for email sent before next update

## Description:
As explained and agreed with @matishaladiwala in https://github.com/mautic/mautic/issues/3461, there was a bug but the main issue was that the way that clicks were calculated was not email standard stats.
You'll now have:

### Email overview

- Emails sent
- Emails read
- :1st_place_medal: **NEW - Emails clicked** > before it was the amount of clicks (weird calculation)
- Emails bounced
- Emails unsubscribed


### Email report

- Emails sent
- Emails sent ratio
- Emails read
- Emails read ratio
- :1st_place_medal: **NEW - Emails clicked**
- :1st_place_medal: **NEW - Emails clicked ratio**

## Steps to test this PR:

1. Apply PR & update schema
2. Create 2 segments with at least 2 contacts inside
3. Create one marketing email, with at least 2 links, both segments as source, send email.
4. Create simple campaign to send an email, with at least 2 links,  with both segments as source, execute campagn.
5. Play with link in receive email

- Check count of click / unique click / email clicked in `emails/view`
- Check stat in report panel
